src/fdc/DiskImageUtils.cc

```diff
@@ -6,11 +6,18 @@
 #include "enumerate.hh"
 #include "random.hh"
 #include "xrange.hh"
+#include <algorithm>
+#include <bit>
 #include <cassert>
 #include <ctime>
 
 namespace openmsx::DiskImageUtils {
 
+static constexpr unsigned MAX_FAT12_CLUSTER_COUNT = 0xFF4;
+static constexpr unsigned MAX_FAT16_CLUSTER_COUNT = 0xFFF4;
```

Comment:
> Just making sure I understand: this value comes from `MAX_FAT - 2` (which implies that MAX_FAT is exclusive (the first value that's no longer a valid cluster number)).

Indeed, where 2 is `FIRST_CLUSTER`.

> Detail: MAX_FAT12_CLUSTER_COUNT is not used. But fine if you leave it for symmetry. Or alternatively you could move `MAX_FAT16_CLUSTER_COUNT` closer to the only place where it's used (inside the then-branch in the `setBootSector()` function, around line 120).

I think instead I’ll add support for the `-nextor` option as `MSXBootSectorType::NEXTOR` (instead of extended), which will use the FAT12 version for volumes smaller than 32 MB.